airflow/models/dag.py

```diff
@@ -690,6 +694,7 @@ def __init__(
                 )
         self.dagrun_timeout = dagrun_timeout
         self.sla_miss_callback = sla_miss_callback
+        self.toggle_dag_callback = toggle_dag_callback
```

Comment:
@potiuk I have added toggle callback in dag and considering if it could be handled like how manage_slas are handled in processor.py by checking if it violates sla and send callback and additional email notification

Do you think this would be the right direction to proceed?

Reply:
@potiuk can you give me your views when you have time?

Reply:
Yes, it could be done this way, though it's a bit complex to implement, just to warn you :)